pkg/monitor/cluster/nodeconditions.go

```diff
@@ -32,9 +36,10 @@ func (mon *Monitor) emitNodeConditions(ctx context.Context) error {
 			}
 
 			mon.emitGauge("node.conditions", 1, map[string]string{
-				"nodeName": n.Name,
-				"status":   string(c.Status),
-				"type":     string(c.Type),
+				"nodeName":     n.Name,
+				"status":       string(c.Status),
+				"type":         string(c.Type),
+				"spotInstance": strconv.FormatBool(mon.isSpotInstance(ctx, n)),
```

Comment:
How many additional API calls are we making with this change for a cluster with 6 nodes (3 master and 3 worker)? Let's assume that that each node has 3 conditions (ready, memory pressure and disk pressure).

Reply:
short answer: too many. I think we should move this api call up a level in the loop. Do we really need to check if a master is a spot instance? that could also save us some calls.

Reply:
I've moved the api call to be one per node instead of one per condition per node.

Reply:
Further refactored to: https://github.com/Azure/ARO-RP/pull/1928#discussion_r798043220